### PR TITLE
DRY: Use utility function to get course info from LMS

### PR DESCRIPTION
**Pull request by mjfrey**
Factor out course API calls as a utility.

### CODE REVIEW

**Comment by mjfrey** [comment]
@vkaracic

File: ecommerce/courses/tests/test_utils.py [hunk 1]
```diff
@@ -1,11 +1,18 @@
+import hashlib
```
Comment (vkaracic) [inline ecommerce/courses/tests/test_utils.py:1]: Watch out for the order of imports, you should add a line here to distinguish between standard library imports and third party ones.

File: ecommerce/courses/tests/test_utils.py [hunk 2]
```diff
@@ -24,3 +31,22 @@ def test_mode_for_seat(self, certificate_type, id_verification_required, mode):
...
+        course_url = get_lms_url('api/courses/v1/courses/{}/'.format(course.id))
+        httpretty.register_uri(httpretty.GET, course_url, status=200, content_type='application/json')
+
+        cache_key = 'courses_api_detail_{}'.format(course.id)
+        cache_hash = hashlib.md5(cache_key).hexdigest()
+        cached_course = cache.get(cache_hash)
+        self.assertIsNone(cached_course)
+
+        response = get_course_info(course.id)
+        self.assertEqual(response, 'HTTPretty :)')
```
Comment (vkaracic) [inline ecommerce/courses/tests/test_utils.py]: You should check the status instead. The default outputs of third party imports (like `HTTPretty :)` here) are subject to change so generally it's better to test for the values that we have a control of, like the status code that you set on line 39, or set a custom body for the registered URI and test it here.

File: ecommerce/courses/tests/test_utils.py [hunk 3]
```diff
@@ -24,3 +31,22 @@ def test_mode_for_seat(self, certificate_type, id_verification_required, mode):
...
+
+        response = get_course_info(course.id)
```
Comment (vkaracic) [inline ecommerce/courses/tests/test_utils.py]: I don't think there is a need to do this twice. I overlooked that there is no `else` part so practically all the lines in the util function will get hit by the test with the first call on line 46.

File: ecommerce/courses/utils.py [hunk 4]
```diff
@@ -8,3 +17,14 @@ def mode_for_seat(seat):
         return 'audit'
 
     return certificate_type
+
+
+def get_course_info(course_key):
```
Comment (vkaracic) [inline ecommerce/courses/utils.py]: You should add a docstring explaining what this function does.

Comment (vkaracic) [inline ecommerce/courses/utils.py]: Also what do you think about giving this a bit more descriptive name, the current one seems a bit too broad. Something like `get_course_info_from_lms`?

**Comment by vkaracic** [comment]
:+1:

File: ecommerce/courses/utils.py [hunk 5]
```diff
@@ -8,3 +17,15 @@ def mode_for_seat(seat):
         return 'audit'
 
     return certificate_type
+
+
+def get_course_info_from_lms(course_key):
```
Comment (clintonb) [inline ecommerce/courses/utils.py:22]: get_course_info is actually fine. This data should be coming from the Course Catalog API.

**Pull request merged**